src/io/flutter/run/LaunchState.java

```diff
@@ -320,16 +334,25 @@ protected final RunContentDescriptor doExecute(@NotNull RunProfileState state, @
       final ProcessHandler process = getRunningAppProcess(launchState.runConfig);
       if (process != null) {
         final FlutterApp app = FlutterApp.fromProcess(process);
-        if (app != null && executorId.equals(app.getMode().mode())) {
-          final FlutterLaunchMode launchMode = FlutterLaunchMode.getMode(env);
-          if (launchMode.supportsReload() && app.isStarted()) {
-            // Map a re-run action to a flutter full restart.
-            FileDocumentManager.getInstance().saveAllDocuments();
-            app.performRestartApp();
-          }
+
+        String selectedDeviceId = null;
+        final FlutterDevice selectedDevice = DeviceService.getInstance(env.getProject()).getSelectedDevice();
+        if (selectedDevice != null) {
+          selectedDeviceId = selectedDevice.deviceId();
```

Comment:
Nit: maybe extract this into a method (`getSelectedDeviceId()`?) and share w/ `canRun(..)`?